### PR TITLE
Rename whitelist -> allowlist and blacklist -> blocklist

### PR DESCRIPTION
**Pull request by hlopko**
To stay backwards compatible:
* For the command line arguments I added aliases to old flag spellings
* For the Builder methods I changed old methods to forward to new ones + marked them as deprecated.

Fixes on https://github.com/rust-lang/rust-bindgen/issues/1812.

### CODE REVIEW

**Comment by hlopko** [comment]
Presubmits failures are (hopefully) being addressed in https://github.com/rust-lang/rust-bindgen/pull/1991.

**Comment by kulp** [comment]
Was your intent to submit a separate PR for `blacklist` -> `blocklist` ? That might be reasonable; it just did not seem explicit, and it would not be desirable to leave things half-done by merging only one such PR.

**Comment by hlopko** [comment]
Oh yes, that was my intent. I'm happy to extend this PR to also cover the blocklist if that's preferable. I also wanted to check in with owners to see if my approach is sane and PR has a chance to be merged.

**Comment by kulp** [comment]
> I'm happy to extend this PR to also cover the blocklist if that's preferable.

I am not personally going to express an opinion in this case -- I just wanted to make sure we captured the approach one way or another. @emilio is the one whose opinions count. Thank you for clarifying, @hlopko.

**Comment by emilio** [comment]
This looks great, thanks! Merging both renames at the same time seems better, at least during the same release. If you want to submit the second patch doing the blacklist rename that'd be great, either on this PR or in another.

I'm very confused about the test failures, master is green, so something is clearly off... Maybe the CI environment changed somehow?

Tests on clang 11 are passing for me on Linux... Are you on mac by any chance?

**Comment by hlopko** [comment]
I've updated this PR to include the blacklist->blocklist change.

I'll try to investigate what is going on with tests. They also fail for me locally. I'm on linux, AFAIK on llvm-11 (but I'll double check).

Thanks for reviewing! :)

**Comment by hlopko** [comment]
With the https://github.com/rust-lang/rust-bindgen/pull/1996 I can run `cargo test` locally and get a fully green run. I still don't know what changed on the CI. Is it possible that https://github.com/rust-lang/rust-bindgen/pull/1996 will fix that too?

**Comment by emilio** [comment]
Can you rebase on top of master to see if that's the case? https://github.com/rust-lang/rust-bindgen/pull/1996 was green on CI so maybe it does :)

**Comment by hlopko** [comment]
It didn't help :(

File: tests/expectations/tests/issue-643-inner-struct.rs [hunk 1]
```diff
@@ -117,6 +117,54 @@ fn bindgen_test_layout_rte_ring() {
         8usize,
         concat!("Alignment of ", stringify!(rte_ring))
     );
+    assert_eq!(
```
Comment (emilio) [inline tests/expectations/tests/issue-643-inner-struct.rs:120]: Yeah, these changes are wrong. They were done with modern clang, but these expectations are for older clang, so I think you just need to remove this change.

Comment (hlopko) [reply tests/expectations/tests/issue-643-inner-struct.rs]: Oh that's embarrassing. Fixed now.

**Comment by hlopko** [comment]
Also renamed test file names.

**Comment by hlopko** [comment]
Friendly ping :)

**Review by emilio — APPROVED** [review]
Yup, sorry, I looked at this while CI was still running and forgot. Thanks a lot for doing this again!

**Pull request merged**